README.md

```diff
@@ -1,95 +1,108 @@
+- Create a global variable called `cslib_ret` to hold return values.
 
-### Usage
+```choicescript
+__startup.txt__
 
-To use cslib, you need only include the cslib_ prefixed scene files in your project, and define a few configuration variables in your startup scene (details further below). You can then call each individual routine as required, via gosub_scene. Documentation on each routine, including its functionality and required parameters (arguments) can be found in the commented source files.
+*create cslib_ret 0
+```
 
-### Limitations
+- That's it! :partying_face:
 
-Sadly, cslib is not always going to be a one-size-fits all solution to your ChoiceScript woes. There are inherent limitations in ChoiceScript that cannot be overcome, no matter how much extra ChoiceScript we write. In addition, when writing a utility like cslib (that aims to appeal to a large number of people), certain tradeoffs have to be made. So, if you plan to use it, you should keep the following limitations in mind:
+You can now call any of the routines described in the file. :muscle:
 
-- **Performance**
-ChoiceScript isn't built for performance, and cslib isn't going to make that any better. The library relies heavily on usage of *gosub_scene, which is costly but it's also the only reliable way to handle complex variable scoping. If you're using the odd function or two as a utility here and there, then you shouldn't have any problems. If, however, you're planning on a super awesome complex simulator that loops through thousands of strings which it passes to cslib:replace or cslib:find... At that point, you might want to consider other options (including a different scripting language!).
+Routines that need to return a value will generally do so through the dedicated global variable: `cslib_ret`. So make sure to capture its value into another variable!
 
-- **\*temp scope**
-Due to the usage of \*gosub_scene, and inherent limitations in ChoiceScript, access to current game state stored in temporary (\*temp) variables from inside cslib functions isn't going to be possible. Most of the time this isn't a problem as you can (and should!) be passing your values as parameters. However, there are definitely some desirable use-cases that aren't currently possible. The best example is passing an "array" (mytemp_1, mytemp_2, mytemp_3) to a function. You can't reliably do this with parameters as you can't change the amount of parameters you're passing at runtime. You can get around it with globals (\*create'd) arrays by passing the variable "prefix" (e.g. 'myvar' in myvar_1, myvar_2) as an argument and de-referencing it via {} syntax (and providing a length). However, having to create every array in the global scope is far from ideal.
+Note that some routines will need you to create more global variables, or follow a naming convention, make sure you read the documentation on the routine before using it.
```

Comment:
ADD: This is true, for instance, for array-based functions.

Reply:
Might be worth adding an extra point on arrays specifically before release, but the README isn't the place for it.
Because we can use Github's wiki functionality to start writing some docs?